src/components/NavMenu.tsx

```diff
@@ -2,138 +2,160 @@ import React, { useState, Fragment } from "react";
 import { NavLink } from "react-router-dom";
 import styled, { css } from "styled-components";
 import { useTranslation } from "react-i18next";
-import menuButton from "./assets/nav-menu/menu-button.png";
 import {
+  color,
   space,
-  SpaceProps,
-  TypographyProps,
   typography,
-  layout,
-  LayoutProps,
   position,
-  PositionProps,
-  color,
+  grid,
+  flexbox,
   ColorProps,
-  border,
-  BorderProps,
+  SpaceProps,
+  TypographyProps,
+  PositionProps,
+  GridProps,
+  FlexboxProps,
 } from "styled-system";
 
-const visibleStyles = css`
+const overlayStyles = css`
   display: flex;
   flex-direction: column;
-  height: 100%;
-  background-color: rgba(0, 0, 0, 0.1);
-  font-size: 36px;
-  justify-content: center;
-  text-align: center;
+  height: 100vh;
   width: 100%;
-  font-family: SangBleu OG Serif Light Regular;
+  justify-content: center;
   border: none;
+  background-image: url("/assets/nav-menu/background-inverted.png");
+  background-size: cover;
+  opacity: 1;
 `;
-const NavMenuOverlay = styled.dialog<{ isOpen: boolean }>`
+
+const Overlay = styled.dialog<{ isOpen: boolean }>`
   display: none;
-  ${(props) => props.isOpen && visibleStyles}
+  ${(props) => props.isOpen && overlayStyles}
 `;
 
-type MenuProps = SpaceProps &
-  TypographyProps &
-  PositionProps &
-  LayoutProps &
-  ColorProps &
-  BorderProps;
-
-const MenuButton = styled.button<MenuProps>`
+const Logo = styled.p<PositionProps & TypographyProps>`
   ${position}
-  ${layout}
-  ${border}
-  ${color}
+  ${typography}
 `;
 
-const MenuItem = styled.li<MenuProps>`
-  ${space}
+const MenuButton = styled.button<
+  TypographyProps & PositionProps & FlexboxProps & SpaceProps
+>`
+  display: grid;
+  border: none;
+  background: transparent;
   ${typography}
-  ${layout}
-  ${color}
-  &:focus,
-  &:hover {
-    font-size: 32px;
+  ${position}
+  ${flexbox}
+  ${space}
+`;
+
+const MenuContainer = styled.div<SpaceProps>`
+  ${space};
+`;
+
+const Menu = styled.ul<GridProps & TypographyProps>`
+  ${grid};
+  ${typography};
+`;
+
+const MenuLink = styled(NavLink)<ColorProps & TypographyProps>`
+  ${color};
+  ${typography};
   }
-  &:visited,
-  &:link,
-  &:active {
-    text-decoration: none;
+`;
+
+const MenuText = styled.li`
+  transition: transform 0.4s;
+  &:hover {
+    transform: scale(1.01);
+    transform-origin: left;
```

Comment:
This line needs to be outside of the hover, otherwise immediately after the hover the transform jumps to the left!

Reply:
yeeeeh